### ISSUE TITLE
improving train data labels using test data labels according to model result

### ISSUE THREAD

**Issue by ilhansevval** [ultralytics/ultralytics#1393]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I want to improve my accuracy during the test images. In my train data, some class too less. So my model couldn't make good predictions to those clases. I want create more labels for training using my test data according to my model result. Do you have any solution of this problem in yolov8. 

Also, I have one more question which is really important. What is exact data transformation format for yolov8. In some articles they taking labels this way:

<class number x1/width of images y1/height of images x2/width of images y2/height of images>

But in some articles they say, 

firstly find the xmin xmax ymin ymax for bounding boxes,

then finding the width and height of bounding boxes ( width = x_max - x_min, height = y_max - y_min )

third step is finding the center of bounding boxes, ( x_center = x_min + width / 2,  y_center = y_min + height / 2 )

finally normalize the coordinates (x_center /= width of images, y_center /= height of images, width /= width of images, 
height /= height of images )

<class number, normalized x center, normalized y center, normalized width of bounding box, normalized height of bounding box> 

I found two different way to transform data labels, If you make certain of this subject, i'll be really pleasure.

Could answer those two questions ?

**Comment by Rusab** [comment]
Hi @ilhansevval, 
1. You can increase the training samples of the under-represented classes via augmentation. Or add more samples of those classes if possible. You shouldn't modify the ground truth labels depending on your models prediction. 

2. The format for YOLO labels are:
<class_number> <normalized_x_center> <normalized_y_center> <normalized_width> <normalized_height>
(the 2nd format you mentioned)

**Comment by ilhansevval** [comment]
But when i use second option models gives error, and won't working. But when i use the first method it works fine. 
How can i create labels after augmentation, for augmented images ? I want to add this images on my training data to improve my model prediction results.

**Comment by glenn-jocher** [comment]
@ilhansevval YOLOv8 automatically augments your training data, there are no additional steps required other than to train. I'll paste our YOLOv5 guide below:

👋 Hello! Thanks for asking about **image augmentation**. YOLOv5 🚀 applies online imagespace and colorspace augmentations in the trainloader (but not the val_loader) to present a new and unique augmented [Mosaic](https://github.com/ultralytics/yolov5/blob/90b7895d652c3bd3d361b2d6e9aee900fd67f5f7/utils/datasets.py#L678-L732) (original image + 3 random images) each time an image is loaded for training. **Images are never presented twice in the same way**.

![YOLOv5 augmentation](https://user-images.githubusercontent.com/26833433/120995721-f3cfed00-c785-11eb-8ee2-b6ef2fa205e8.jpg)

### Augmentation Hyperparameters

The hyperparameters used to define these augmentations are in your hyperparameter file (default `data/hyp.scratch.yaml`) defined when training:
```
python train.py --hyp hyp.scratch-low.yaml
```
https://github.com/ultralytics/yolov5/blob/b94b59e199047aa8bf2cdd4401ae9f5f42b929e6/data/hyps/hyp.scratch-low.yaml#L6-L34

### Augmentation Previews

You can view the effect of your augmentation policy in your train_batch*.jpg images once training starts. These images will be in your train logging directory, typically `yolov5/runs/train/exp`:

`train_batch0.jpg` shows train batch 0 mosaics and labels:  
<img src="https://user-images.githubusercontent.com/26833433/131255960-b536647f-7c61-4f60-bbc5-cb2544d71b2a.jpg" width="800">


### YOLOv5 Albumentations Integration

[YOLOv5](https://github.com/ultralytics/yolov5) 🚀 is now fully integrated with [Albumentations](https://github.com/albumentations-team/albumentations), a popular open-source image augmentation package. Now you can train the world's best Vision AI models even better with custom Albumentations 😃! 

PR https://github.com/ultralytics/yolov5/pull/3882 implements this integration, which will automatically apply Albumentations transforms during YOLOv5 training if `albumentations>=1.0.3` is installed in your environment. See https://github.com/ultralytics/yolov5/pull/3882 for full details.

Example `train_batch0.jpg` on COCO128 dataset with Blur, MedianBlur and ToGray. See the YOLOv5 [Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb) to reproduce: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

<img src="https://user-images.githubusercontent.com/26833433/124400879-ff331b80-dd25-11eb-9b67-fe85ac4ca104.jpg" width="800">

Good luck 🍀 and let us know if you have any other questions!

**Comment by Laughing-q** [comment]
@ilhansevval Are you trying to train a segmentation model or a detection model?
for segmentation model:
the data format should be just like your first format:
```
class_id x1 y1 x2 y2 ... xn yn
```
for detection model it should be like the second one:
```
class_id cx cy w h
```
all the coordinates should be normalized.
BTW we'll add a section about how to prepare your data format in our docs soon. :)
